src/Widgets/calendar/GridDay.vala

```diff
@@ -115,6 +115,8 @@ public class DateTime.Widgets.GridDay : Gtk.EventBox {
                         event_dots[event_uid] = event_dot;
 
                         event_grid.add (event_dot);
+                        grid.attach (event_grid, 0, 1);
```

Comment:
You're calling attach several times here, you can check `if (event_grid.get_parent () != null)` before calling it

Reply:
Done. See the newest commit. :smile: